### PR TITLE
Docs: Change formatting for sequence of steps

### PR DESCRIPTION
**Pull request by vivganes**
A numbered list is used when the sequence of steps matters. A bulleted list is used to denote a set of items where the sequence does not matter. 
 
The workflow is a sequence and hence changed to a numbered list.

Reference: https://www.w3.org/wiki/HTML_lists#Ordered_lists

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,9 +24,9 @@ facilitates social contribution, easy testing and peer review.
 
 To contribute a patch, the workflow is as follows:
 
-  - Fork repository
-  - Create topic branch
-  - Commit patches
+  1. Fork repository
```
Comment (promag) [inline CONTRIBUTING.md]: Not sure what others think, but since it's markdown, all items can be ` 1. `.

Comment (vivganes) [reply CONTRIBUTING.md]: @promag Thanks for the suggestion! I just changed the same to suggested markdown format.

**Comment by MarcoFalke** [comment]
Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

**Comment by vivganes** [comment]
@MarcoFalke Done!

**Pull request merged**